### ISSUE TITLE
CA2021 False positive for nullable tuples with named fields

### ISSUE THREAD

**Issue by RReverser** [dotnet/roslyn-analyzers#7407]
### Analyzer

**Diagnostic ID**: CA2021: Don't call Enumerable.Cast<T> or Enumerable.OfType<T> with incompatible types

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: 8.0.400

### Steps To Reproduce

MRE:

```csharp
using Tuple = (int a, int b);

Tuple[] arr = [(1, 2)];

foreach (var maybeNull in arr.Cast<Tuple?>())
{
    Console.WriteLine(maybeNull is { } nonNull ? $"{nonNull.a}, {nonNull.b}" : "null");
}
```

### Expected behavior

No warning, because this code runs successfully and is simply casting a tuple to its nullable wrapper, which is a valid cast.

### Actual behavior

Getting `warning CA2021: Type '(int a, int b)' is incompatible with type '(int a, int b)?' and cast attempts will throw InvalidCastException at runtime`.

## Additional context

Oddly, the warning does not show up when casting to the same tuple but without compile-time names (`arr.Cast<(int, int)?>()`) even though they're equivalent in runtime.

**Comment by sukreshmanda** [comment]
I tried to reproduce this in 9.* version but not able to. Looks like this works on 8.* or below

**Comment by fowl2** [comment]
I also can't reproduce this in version 9. Pretty sure it was fixed by https://github.com/dotnet/roslyn-analyzers/pull/7183.

@RReverser are you still experiencing this?

**Comment by RReverser** [comment]
Sorry, I'm not working much with .NET anymore, but if you believe it was fixed, feel free to close the issue.

**Comment by RReverser** [comment]
I'll close it, if anyone else runs into one, I'm sure they can open a new issue.
